astropy/io/fits/column.py

```diff
@@ -2402,3 +2414,105 @@ def _convert_ascii_format(format, reverse=False):
             recformat += str(width)
 
         return recformat
+
+
+def _parse_tdisp_format(tdisp):
+    """
+    Parse the ``TDISPn`` keywords for ASCII and binary tables into a
+    ``(format, width, precision, exponential)`` tuple (the TDISP values
+    for ASCII and binary are identical except for 'Lw', which is only present in BINTABLE
+    extensions
+    """
+
+    # Use appropriate regex for format type
+    tdisp = tdisp.strip()
+    if tdisp[0] == 'E':
```

Comment:
With a more complete dict this would become:
```
fmt_key = tdisp[0] if tdisp[0] != 'E' or tdisp[1] not in 'NS' else tdisp[:2]
try:
    tdisp_re = TDISP_RE_DICT[fmt_key]
except KeyError:
    raise VerifyError('...')
```